src/test/scala/edu/ie3/simona/agent/grid/GridAgentSetup2WSpec.scala

```diff
@@ -6,44 +6,40 @@
 
 package edu.ie3.simona.agent.grid
 
-import org.apache.pekko.actor.{
-  Actor,
-  ActorIdentity,
-  ActorRef,
-  ActorSystem,
-  Identify,
-  Props,
-}
-import org.apache.pekko.testkit.ImplicitSender
-import org.apache.pekko.util.Timeout
-import com.typesafe.config.ConfigFactory
 import com.typesafe.scalalogging.LazyLogging
 import edu.ie3.datamodel.models.result.ResultEntity
 import edu.ie3.simona.agent.EnvironmentRefs
+import edu.ie3.simona.agent.grid.GridAgentMessage.StringAdapter
 import edu.ie3.simona.io.result.ResultSinkType
 import edu.ie3.simona.sim.setup.SimonaStandaloneSetup
+import edu.ie3.simona.test.common.ConfigTestData
 import edu.ie3.simona.test.common.input.TransformerInputTestData
-import edu.ie3.simona.test.common.{ConfigTestData, TestKitWithShutdown}
 import edu.ie3.simona.util.ResultFileHierarchy
 import edu.ie3.simona.util.ResultFileHierarchy.ResultEntityPathConfig
+import org.apache.pekko.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import org.apache.pekko.actor.typed.receptionist.Receptionist.{
+  Listing,
+  Register,
+  Subscribe,
+}
+import org.apache.pekko.actor.typed.receptionist.ServiceKey
+import org.apache.pekko.actor.typed.scaladsl.AskPattern.Askable
+import org.apache.pekko.actor.typed.scaladsl.Behaviors
+import org.apache.pekko.actor.typed.scaladsl.adapter.{
+  TypedActorContextOps,
+  TypedActorRefOps,
+}
+import org.apache.pekko.actor.typed.{ActorRef, Scheduler}
+import org.apache.pekko.actor.{ActorRef => classicRef}
+import org.apache.pekko.util.Timeout
 import org.scalatest.wordspec.AnyWordSpecLike
 
 import java.util.concurrent.TimeUnit
 import scala.concurrent.Await
 import scala.concurrent.duration.Duration
 
 class GridAgentSetup2WSpec
```

Comment:
I think this would be a perfect example for using [synchronous testing](https://nightlies.apache.org/pekko/docs/pekko/1.0/docs/typed/testing-sync.html). Hit me up if we should look at this together

Reply:
To be more specific, instead of trying to find the GridAgent actors via search/selection asynchronously (`actorSelection` etc.), when you synchronously test a behavior, testing whether an actor has spawned becomes really easy:

`testKit.expectEffect(Spawned(childActor, "child"))`

or

`testKit.expectEffect(SpawnedAnonymous(childActor))`